raqm.c

```diff
@@ -649,14 +656,14 @@ raqm_shape_u32 (unsigned int* u32_str,
     {
         hb_glyph_info = hb_buffer_get_glyph_infos (runs[i].hb_buffer, &glyph_count);
         hb_glyph_position = hb_buffer_get_glyph_positions (runs[i].hb_buffer, &postion_length);
-        int j;
-        for (j = 0; j < glyph_count; j++)
+
+        for (k = 0; k < glyph_count; k++)
         {
-            glyph_info[index].index = hb_glyph_info[j].codepoint;
-            glyph_info[index].x_offset = hb_glyph_position[j].x_offset;
-            glyph_info[index].y_offset = hb_glyph_position[j].y_offset;
-            glyph_info[index].x_advance = hb_glyph_position[j].x_advance;
-            glyph_info[index].cluster = hb_glyph_info[j].cluster;
+            glyph_info[index].index = (int)(hb_glyph_info[k].codepoint);
```

Comment:
I think `index` should be declared as `unsigned int` which would avoid the need of the cast (and is the correct thing to do).